api/pkg/service/resource/resource.go

```diff
@@ -350,7 +352,30 @@ func tinyVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
 	return res
 }
 
-func minVersionInfo(r model.ResourceVersion) *resource.ResourceVersionData {
+// This function finds the resource details using resource id
+// and returns a HubURL struct which has resourceName, kind of
+// the resource and the catalog from which the resource belongs
+func resourceHubUrlData(resId uint, req res.Request) (res.HubUrl, error) {
+	var resource model.Resource
+
+	q := req.Db.Model(&model.Resource{}).Where("id = ?", resId).Preload("Catalog")
```

Comment:
we are still doing query to fetch details?
the reason we created an separate package was to move all database interaction to it
https://github.com/tektoncd/hub/blob/main/api/pkg/shared/resource/resource.go
and consumer would just use the data in a way it wants
so we do shouldn't do a db query in this package
instead of this we preload the details in the query from where data is coming

Reply:
so in this package using `nested preload` for the query your data is coming from wouldn't work?